### ISSUE TITLE
Type is unknown to the importer module with strange workarounds

### ISSUE THREAD

**Issue by wnats** [pschachte/wybe#418]
`d.wybe`
```
pub constructor empty 

pub type a is address {}

pub type c is address {}
```
`b.wybe`
```
use d

?b:d= empty:d
```
Compiling `b.wybe` yields:
```
Error detected during type checking of module(s) b
In unification, unknown type d
test/test/b.wybe:3:2: In unification, unknown type d
test/test/b.wybe:3:8: In unification, unknown type d
```
The issue could be avoided by one of the following:
1. Change the visibility of one of `a` or `c` to private
2. Remove one of `a` or `c`
3. Rename either `a`, `c` or `d`.

**Comment by jimbxb** [comment]
I have trawled through the logs, but this is bizarre. 

Two things that stick out as really odd are:
- changing the names if the types effects this -- that makes me think there's some caching issue, where some outdated .o file is being used
- the lack of location on the first type error -- I wonder where that is generated, and why it doesn't have a location.

**Comment by pschachte** [comment]
Fixed by #443.  Turned out to be a mangled Map that had multiple copies of the same key.
